core/src/main/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCO.scala

```diff
@@ -93,107 +91,179 @@ class RecordInsightsLOCO[T <: Model[T]]
     case m => toOPUnchecked(m).transformFn
   }
   private val labelDummy = RealNN(0.0)
+  private lazy val histories = OpVectorMetadata(getInputSchema()(in1.name)).getColumnHistory()
+  private lazy val featureInfo = histories.map(_.toJson(false))
 
-  private lazy val featureInfo = OpVectorMetadata(getInputSchema()(in1.name)).getColumnHistory().map(_.toJson(false))
+  /**
+   * These are the name of the stages we want to perform an aggregation of the LOCO results over derived features
+   */
+  private val smartTextClassName = classOf[SmartTextVectorizer[_]].getSimpleName
+  private val smartTextMapClassName = classOf[SmartTextMapVectorizer[_]].getSimpleName
+  // Indices of features derived from SmartText(Map)Vectorizer
+  private lazy val textFeatureIndices = histories
+    .filter(_.parentFeatureStages.exists(s => s.contains(smartTextClassName) || s.contains(smartTextMapClassName)))
+    .map(_.index)
+    .distinct.sorted
```

Comment:
Do we need `distinct` here ?

Reply:
Added by @tovbinm. Not sure it is necessary but it seems safe.

Reply:
no need to keep duplicate indices, so added distinct. sorting was added for convenience.

Reply:
Is there a scenario where there can be duplicates ? I don't think so, but yeah it seems safe to add.

Reply:
Afaik, we can't currently have duplicate indices here but these are usually small (~few thousand element) arrays so no biggie.